tests/contributing.py

```diff
@@ -121,3 +126,65 @@ def validate_transformer_integration(transformer):
     print(tabulate(summary, headers='keys', showindex=False))
 
     return validation_error is None and error_trace is None
+
+
+def validate_transformer_performance(desired_transformer):
+    """Validate the performance of a transformer.
+
+    Run the specified Transformer on all the Dataset Generators of the indicated data type
+    and produce a report about its performance and how it compares to the other
+    Transformers of the same data type.
+
+    Args:
+        desired_transformer (string or rdt.transformers.BaseTransformer):
+            The transformer to validate.
+
+    Returns:
+        pandas.DataFrame:
+            Performance results of the transformer.
+    """
+    if isinstance(desired_transformer, str):
+        desired_transformer = get_class(desired_transformer)
+
+    print(f'Validating Performance for transformer {desired_transformer.__name__}\n')
+
+    data_type = desired_transformer.get_input_type()
+    transformers = get_transformers_by_type().get(data_type, [])
+    dataset_generators = get_dataset_generators_by_type().get(data_type, [])
+
+    total_results = pd.DataFrame(
+        [[test_name, 0, 0, True] for test_name in TEST_NAMES],
+        columns=['Test', 'Value', 'Total', 'Valid'],
+    )
+
+    for transformer in transformers:
+        for dataset_generator in dataset_generators:
+            results = {}
+            validate_performance(transformer, dataset_generator, results=results)
+
+            for test in TEST_NAMES:
+                value, valid = results[test]
+                total_results.loc[total_results.Test == test, 'Total'] += value
+                total_results.loc[total_results.Test == test, 'Valid'] &= valid
+
+                if transformer == desired_transformer:
+                    total_results.loc[total_results.Test == test, 'Value'] += value
+
+    if total_results['Valid'].all():
+        print('SUCCESS: The Performance Tests were successful.')
+    else:
+        print('ERROR: One or more Performance Tests were NOT successful.')
+
+    final_results = pd.DataFrame()
+    final_results['Evaluation Metric'] = total_results['Test']
+    final_results['Value'] = total_results['Value'] / len(dataset_generators)
+    final_results['Units'] = np.where(
+        final_results['Evaluation Metric'].str.contains('Time'),
+        's / row',
+        'Mb / row',
+    )
+    final_results['Acceptable'] = np.where(total_results['Valid'], 'Yes', 'No')
+    average = total_results['Total'] / (len(transformers) * len(dataset_generators))
```

Comment:
The average should be computed without including the current transformer.
If there are no other transformers, the `Compared to average` column can be `N/A` or `np.nan`